### PR TITLE
SONAR-14822 Provide a GitHub Action to scan a project

### PR DESCRIPTION
**Pull request by wouter-admiraal-sonarsource**
You can see it in action here: https://github.com/wouter-admiraal-sonarsource/whoswho/blob/master/.github/workflows/build.yml

You can use `wouter-admiraal-sonarsource/sonarqube-scan-action` in your workflows to try it out.

 The run was successful:
 - Logs here: https://github.com/wouter-admiraal-sonarsource/whoswho/runs/2555303817?check_suite_focus=true
 - Analysis result (on SonarCloud for simplicity) here: https://sonarcloud.io/summary/overall?id=whoswho

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+FROM sonarsource/sonar-scanner-cli:4.6
+
+LABEL version="1.0.0" \
+      repository="https://github.com/sonarsource/sonarqube-scan-action" \
+      homepage="https://github.com/sonarsource/sonarqube-scan-action" \
+      maintainer="SonarSource" \
+      com.github.actions.name="SonarQube Scan" \
+      com.github.actions.description="Scan your code with SonarQube to detect Bugs, Vulnerabilities and Code Smells in more than 20 programming languages!" \
```
Comment (philippe-perrin-sonarsource) [inline Dockerfile:8]:
```suggestion
      com.github.actions.description="Scan your code with SonarQube to detect Bugs, Vulnerabilities and Code Smells in more than 27 programming languages!" \
```

File: README.md [hunk 2]
```diff
@@ -1 +1,104 @@
-# sonarqube-scan-action
+# Scan your code with SonarQube
+
+Using this GitHub Action, scan your code with [SonarQube](https://www.sonarqube.org/) to detects Bugs, Vulnerabilities and Code Smells in more than 20 programming languages!
```
Comment (philippe-perrin-sonarsource) [inline README.md:3]:
```suggestion
Using this GitHub Action, scan your code with [SonarQube](https://www.sonarqube.org/) to detects Bugs, Vulnerabilities and Code Smells in more than 27 programming languages!
```

File: action.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+name: SonarQube Scan
+description: >
+  Scan your code with SonarQube to detect Bugs, Vulnerabilities and Code Smells in more than 20 programming languages!
```
Comment (philippe-perrin-sonarsource) [inline action.yml:3]:
```suggestion
  Scan your code with SonarQube to detect Bugs, Vulnerabilities and Code Smells in more than 27 programming languages!
```

File: README.md [hunk 4]
```diff
@@ -1 +1,104 @@
...
+## Do not use this GitHub action if you are in the following situations
+
+* Your code is built with Maven. Read the documentation about our [Scanner for Maven](https://docs.sonarqube.org/latest/analysis/scan/sonarscanner-for-maven/).
+* Your code is built with Gradle. Read the documentation about our [Scanner for Gradle](https://docs.sonarqube.org/latest/analysis/scan/sonarscanner-for-gradle/).
+* You want to analyze a .NET solution. Read the documentation about our [Scanner for .NET](https://docs.sonarqube.org/latest/analysis/scan/sonarscanner-for-msbuild/).
+* You want to analyze C/C++ code. Read the documentation on [analyzing C/C++ code](https://docs.sonarqube.org/latest/analysis/languages/cfamily/).
+
+## Have question or feedback?
+
+To provide feedback (requesting a feature or reporting a bug) please post on the [SonarSource Community Forum](https://community.sonarsource.com/) with the tag `sonarqube-team`.
```
Comment (philippe-perrin-sonarsource) [inline README.md:98]:
```suggestion
To provide feedback (requesting a feature or reporting a bug) please post on the [SonarSource Community Forum](https://community.sonarsource.com/tags/c/help/sq/github-actions).
```

File: entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  echo "This GitHub Action requires the SONAR_TOKEN env variable."
+  exit 1
+fi
+
+if [[ -z "${SONAR_HOST_URL}" ]]; then
+  echo "This GitHub Action requires the SONAR_HOST_URL env variable."
+  exit 1
+fi
+
+if [[ -f "${INPUT_PROJECTBASEDIR%/}pom.xml" ]]; then
```
Comment (philippe-perrin-sonarsource) [inline entrypoint.sh:15]: I'm wondering how necessary this is? We might want to do this in bitbucket pipes too to be consistent.

File: README.md [hunk 6]
```diff
@@ -1 +1,104 @@
...
+- `SONAR_HOST_URL` – **Required** this tells the scanner where SonarQube is hosted. You can set the `SONAR_HOST_URL` environment variable in the "Secrets" settings page of your repository.
+- *`GITHUB_TOKEN` – Provided by Github (see [Authenticating with the GITHUB_TOKEN](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/authenticating-with-the-github_token)).*
+
+## Example of pull request analysis
+
+<img src="./images/SonarQube-analysis-in-Checks.png">
+
+## Do not use this GitHub action if you are in the following situations
+
+* Your code is built with Maven. Read the documentation about our [Scanner for Maven](https://docs.sonarqube.org/latest/analysis/scan/sonarscanner-for-maven/).
```
Comment (philippe-perrin-sonarsource) [inline README.md:91]:
```suggestion
* Your code is built with Maven. Read the documentation about our [Scanner for Maven](https://redirect.sonarsource.com/doc/install-configure-scanner-maven.html).
```

File: README.md [hunk 7]
```diff
@@ -1 +1,104 @@
...
+* Your code is built with Gradle. Read the documentation about our [Scanner for Gradle](https://docs.sonarqube.org/latest/analysis/scan/sonarscanner-for-gradle/).
```
Comment (philippe-perrin-sonarsource) [inline README.md:92]:
```suggestion
* Your code is built with Gradle. Read the documentation about our [Scanner for Gradle](https://redirect.sonarsource.com/doc/gradle.html).
```

File: README.md [hunk 8]
```diff
@@ -1 +1,104 @@
...
+* You want to analyze a .NET solution. Read the documentation about our [Scanner for .NET](https://docs.sonarqube.org/latest/analysis/scan/sonarscanner-for-msbuild/).
```
Comment (philippe-perrin-sonarsource) [inline README.md:93]:
```suggestion
* You want to analyze a .NET solution. Read the documentation about our [Scanner for .NET](https://redirect.sonarsource.com/doc/install-configure-scanner-msbuild.html).
```

File: README.md [hunk 9]
```diff
@@ -1 +1,102 @@
...
+      -Dsonar.python.coverage.reportPaths=coverage.xml
+      -Dsonar.sources=lib/
+      -Dsonar.test.exclusions=tests/**
+      -Dsonar.tests=tests/
+      -Dsonar.verbose=true
+```
+
+More information about possible analysis parameters can be found in [the documentation](https://docs.sonarqube.org/latest/analysis/analysis-parameters/).
+
+### Secrets
```
Comment (jacek-poreda-sonarsource) [inline README.md:78]: Let's name it Environment variables

**Review by jacek-poreda-sonarsource — APPROVED** [review]
LGTM 👍

File: README.md (hunk 9 above)
Comment (wouter-admiraal-sonarsource) [reply README.md]: 👍

**Pull request merged**